libs/langgraph/src/pregel/call.ts

```diff
@@ -0,0 +1,63 @@
+import { RunnableConfig, RunnableSequence } from "@langchain/core/runnables";
+import { AsyncLocalStorageProviderSingleton } from "@langchain/core/singletons";
+import { CONFIG_KEY_CALL, RETURN, TAG_HIDDEN } from "../constants.js";
+import { ChannelWrite, PASSTHROUGH } from "./write.js";
+import { RetryPolicy } from "./utils/index.js";
+import { Promisified } from "./types.js";
+import { RunnableCallable, type RunnableCallableArgs } from "../utils.js";
+/**
+ * Cache for storing runnable sequences to avoid recreating them
+ */
+
+/**
+ * Get a runnable sequence for a function that wraps it in a sequence with a RETURN channel write
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getRunnableForFunc<FuncT extends (...args: any[]) => any>(
+  name: string,
+  func: FuncT
+): RunnableSequence<Parameters<FuncT>, ReturnType<FuncT>> {
+  const run = new RunnableCallable<Parameters<FuncT>, ReturnType<FuncT>>({
+    func: (input: Parameters<FuncT>) => func(...input),
+    name,
+    trace: false,
+    recurse: true,
+  } as RunnableCallableArgs);
+
+  // writes to return channel
+  const seq = new RunnableSequence<Parameters<FuncT>, ReturnType<FuncT>>({
+    name,
+    first: run,
+    last: new ChannelWrite<ReturnType<FuncT>>(
+      [{ channel: RETURN, value: PASSTHROUGH }],
+      [TAG_HIDDEN]
+    ),
+    // TODO: add trace_inputs for task?
+  });
+
+  // don't cache anonymous functions
+  return seq;
+}
+
+export type CallWrapperOptions<FuncT extends (...args: unknown[]) => unknown> =
+  {
+    func: FuncT;
+    name: string;
+    retry?: RetryPolicy;
+  };
+
+export function call<FuncT extends (...args: unknown[]) => unknown>(
+  { func, name, retry }: CallWrapperOptions<FuncT>,
+  ...args: Parameters<FuncT>
+): Promisified<FuncT> {
+  const config =
+    AsyncLocalStorageProviderSingleton.getRunnableConfig() as RunnableConfig;
+  // TODO: type the CONFIG_KEY_CALL function
```

Comment:
Just make sure types for this don't appear in autocomplete for user

Reply:
Yeah, this is for internal devX. Gist is that we're cruisin' for a bruisin' with this passing scheme, as everything is implicit `any` here, which makes things hard to test & maintain. Also makes it less discoverable for external contributors. I won't go too crazy with it in this PR, but my plan is to clean it up some so these keys are typed properly at least for the ones I'm touching here.

Reply:
saving this one for later - needs a bigger refactor than I wanted to add to this already huge PR.